src/Sylius/Bundle/LocaleBundle/Controller/LocaleController.php

```diff
@@ -65,6 +65,6 @@ protected function getLocaleContext()
      */
     protected function getLocaleProvider()
     {
-        return $this->get('sylius.locale_provider');
+        return $this->get('sylius.channel_aware_locale_provider');
```

Comment:
Locale bundle should not know about channels. This should be core specific customization.